pkg/monitoring/observability/rules/alerts/alerts.go

```diff
@@ -0,0 +1,47 @@
+package alerts
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	defaultRunbookURLTemplate = "https://kubevirt.io/monitoring/runbooks/%s"
+	runbookURLTemplateEnv     = "RUNBOOK_URL_TEMPLATE"
+)
+
+func Register(operatorRegistry *operatorrules.Registry) error {
+	alerts := [][]promv1.Rule{
+		clusterAlerts(),
+	}
+
+	runbookURLTemplate := getRunbookURLTemplate()
+	for _, alertGroup := range alerts {
+		for _, alert := range alertGroup {
+			alert.Labels["kubernetes_operator_part_of"] = "kubevirt"
+			alert.Labels["kubernetes_operator_component"] = "cnv-observability"
+			alert.Annotations["runbook_url"] = fmt.Sprintf(runbookURLTemplate, alert.Alert)
+		}
+
+	}
+
+	return operatorRegistry.RegisterAlerts(alerts...)
+}
+
+func getRunbookURLTemplate() string {
+	runbookURLTemplate, exists := os.LookupEnv(runbookURLTemplateEnv)
+	if !exists {
+		runbookURLTemplate = defaultRunbookURLTemplate
+	}
+
+	if strings.Count(runbookURLTemplate, "%s") != 1 {
+		panic(errors.New("runbook URL template must have exactly 1 %s substring"))
```

Comment:
why panic here? this is a running server - and we should not panic from it. This function is called from Register that returns error, so we can return error here to then be returned from Register.

Reply:
removed